uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestDeletionTest.php

```diff
@@ -71,6 +71,43 @@ public function testCanDeleteAPublicHolidayLeaveRequestForASingleContact() {
     $this->assertEquals(0, LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($periodEntitlement));
   }
 
+  public function testDeleteForContactWithSoftDeleteAsTrue() {
+    $absencePeriod = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2017-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2017-12-31')
+    ]);
+    $periodEntitlement = $this->createLeavePeriodEntitlementMockForBalanceTests(
+      new DateTime('2017-01-01'),
+      new DateTime('2017-12-31')
+    );
+    $periodEntitlement->contact_id = 2;
+    $periodEntitlement->type_id = $this->absenceType->id;
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $periodEntitlement->contact_id],
+      ['period_start_date' => $absencePeriod->start_date]
+    );
+
+    $publicHoliday = $this->instantiatePublicHoliday('2017-01-01');
```

Comment:
In a few other tests you did the instantiation manually. How about extracting this `instantiatePublicHoliday` method to a trait so that it can be easily reused in these places where you're doing things manually?